app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -3,9 +3,30 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
 
   require_authentication :all, scopes: [:project]
   schema_type :json_schema
+  before_action :require_project_ownership, only: [:update_project_settings]
+
+  def update_project_settings
+    upp = UserProjectPreference.find_or_initialize_by(project: project, user_id: api_user.user)
+    upp.settings.merge! params[:settings]
```

Comment:
you have to user `params_for` to make the json schema validator run and validate the schema